### PR TITLE
Improve handling of mixed number types

### PR DESCRIPTION
**Pull request by henriknyman**
- Add type coercion from CTNumber to CTFloat so that math functions
  can be used on collections with mixed number types

### CODE REVIEW

File: community/cypher/frontend-3.0/src/main/scala/org/neo4j/cypher/internal/frontend/v3_0/symbols/NumberType.scala [hunk 1]
```diff
@@ -23,6 +23,7 @@ object NumberType {
   val instance = new NumberType() {
     val parentType = CTAny
     override val isAbstract = true
+    override lazy val coercibleTo: Set[CypherType] = Set(CTFloat)
```
Comment (henriknyman) [inline community/cypher/frontend-3.0/src/main/scala/org/neo4j/cypher/internal/frontend/v3_0/symbols/NumberType.scala]: This is still up to debate...

**Comment by pontusmelke** [comment]
if you add `fixes #5565` to the commit message it will automatically close the issue when merged

**Comment by pontusmelke** [comment]
I think you need to rebase to get rid of the build error

File: community/cypher/frontend-3.0/src/main/scala/org/neo4j/cypher/internal/frontend/v3_0/ast/functions/ToFloat.scala [hunk 2]
```diff
@@ -28,6 +28,7 @@ case object ToFloat extends Function with SimpleTypedFunction {
   val signatures = Vector(
     Signature(argumentTypes = Vector(CTFloat), outputType = CTFloat),
     Signature(argumentTypes = Vector(CTInteger), outputType = CTFloat),
-    Signature(argumentTypes = Vector(CTString), outputType = CTFloat)
+    Signature(argumentTypes = Vector(CTString), outputType = CTFloat),
+    Signature(argumentTypes = Vector(CTNumber), outputType = CTFloat)
```
Comment (cleishm) [inline community/cypher/frontend-3.0/src/main/scala/org/neo4j/cypher/internal/frontend/v3_0/ast/functions/ToFloat.scala]: If you add CTNumber, you don't need CTFloat and CTInteger signatures.

**Comment by henriknyman** [comment]
Rebased and removed the fix for toInt() and toFloat() (which is now merged in 3.0 through a separate PR)

**Comment by henriknyman** [comment]
This PR needs to wait for a decision by the Cypher Language Group (and may be abandoned)

**Comment by Mats-SX** [comment]
Closing this until decision has been made.

**Pull request closed without merging**